File: testdata/fixtures/ai/csharp/detect-openai.cs
// AI usage detection: OpenAI ChatClient
// ruleid: detect-openai
using OpenAI;
using System;
// ruleid: detect-openai
var client = new ChatClient("gpt-4o", "key");
// ruleid: detect-openai
var result = client.CompleteChat("Hello");
// ok: detect-openai
using System;
// ok: detect-openai
namespace Demo {}
// ok: detect-openai
class Safe {}

File: testdata/fixtures/ai/typescript/detect-openai.ts
// AI usage detection: OpenAI
// ruleid: detect-openai
import "openai";
// ruleid: detect-openai
import OpenAI from "openai";
// ruleid: detect-openai
const client = new OpenAI({ apiKey: "key" });
// ruleid: detect-openai
const r = client.chat.completions.create({ model: "gpt-4", messages: [] });
// ok: detect-openai
import fs from "fs";
// ok: detect-openai
const text = "plain string";
// ok: detect-openai
console.log("hello");
